src/DateTimePicker.jsx

```diff
@@ -132,6 +132,7 @@ class DateTimePicker extends React.Component {
 	*/
 	setDateTime(value) {
 		this.setState({ datetime: new Date(value) });
+		this.props.onChangeCallback(this.state.datetime, this.props.name);
```

Comment:
this function name and signature are confusing

1. it's called `onChange...`, but it's called from `setDateTime`, which can technically be called at any time, not just when the input changes
2. `onChangeCallback` and `name` are supplied as props, which means that the parent already knows the value of `name` and shouldn't need for it to be returned as an argument. You can use currying to set up an `onChangeCallback` function that only relies on `this.state.datetime`, e.g.

    ```jsx
    // get a function that can use `name` without needing it as a new arg
    const getOnChangeCallback = name => datetime => { ... }; 
    const onChangeCallback = getOnChangeCallback(name);

    <DateTimePicker onChangeCallback={onChangeCallback} name={name} ... />
    ```

It's also a little strange that `onChangeCallback` doesn't take an `event` argument the way most `onX` event handlers do, but I see that convention isn't being used for any of the other form components at the moment, so probably no biggie